src/UiPath.Workflow.Runtime/ActivityInstance.cs

```diff
@@ -402,7 +402,16 @@ internal string OwnerName
     }
 
     [DataMember(EmitDefaultValue = false)]
-    public Version ImplementationVersion { get; internal set; }
+    public Version ImplementationVersion { get; internal set; }
+
+    public void AddAutomationTrackerId()
+    {
+        var properties = new ExecutionProperties(null, this, PropertyManager);
+        var existing = properties.Find("BPOId") as string;
+        properties.Add("BPOId", (existing + "-" + Guid.NewGuid().ToString("N")).Trim('-'), true, false);
```

Comment:
`.` is the separator for activityIds, so I was thinking to use the same?

Reply:
done